providers/themeprovider.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { createContext, useEffect, useState } from 'react';
```

Comment:
What about using this? https://www.npmjs.com/package/next-themes, sounds like a good abstraction layer for the theme provider?

Reply:
Let me look into it

Reply:
`next-themes` is changing the color scheme and data-theme attribute of HTML only. We have handled all the dark theme styles with `body` tag with `.dark` className.

if we want to use `next-themes`  either we update the body tag with dark/light className when theme is getting toggled or change the existing stylesheet to select theme from `data-theme` attribute. Also I don't see a problem with current theme provider which suits our need.

Reply:
> if we want to use next-themes either we update the body tag with dark/light className when theme is getting toggled or change the existing stylesheet to select theme from data-theme attribute. Also I don't see a problem with current theme provider which suits our need.

I think that's fine, honestly speaking. Let's switch to next-themes :)